### ISSUE TITLE
Use of -Q 0 in bcftools call

### ISSUE THREAD

**Issue by tseemann** [jodyphelan/TBProfiler#97]
I note you use `-Q 0` in bcftools in your variant calling. This means to use read alignment with mapping quality 0, which means they were multi-mapping reads. 

This is normally risky, but maybe you allow it because you are only calling variants in specific regions and you know that there is no multimapping issues within those regions?

But does that mean your mixed alleles proportions will be wrong?

For BWA MEM and minimap2  mapqual=60 means uniquely mapping, and i think its 42 for bowtie2.

**Comment by jodyphelan** [comment]
Hmmm... I had put `-Q0` in there for reasoning which I have now forgotten and may be

**Comment by jodyphelan** [comment]
This has now been commited f502c4a

**Comment by jodyphelan** [comment]
Actually... just realised it should be `-q0` rather than `-Q0` that should removed

**Comment by jodyphelan** [comment]
Fixed now in 959e65d

**Comment by tseemann** [comment]
Luckily -q0 was default. Do you have a test suite you re-run to see how these changes affect results?

**Comment by jodyphelan** [comment]
The `-q8` originally came from guys a nanopore who were recommending this for their data. I've got the ~16000 isolates I used for the original benchmarking. I'll do a little analysis of the parameters and see how it affects the results. Might take me a bit... so in the meantime do you think the current parameters are sensible?

**Comment by tseemann** [comment]
Just to confirm we aren't getting mixed up between BASEQ and MAPQ?

```
bcftools mpileup
  -q, --min-MQ INT        skip alignments with mapQ smaller than INT [0]
  -Q, --min-BQ INT        skip bases with baseQ/BAQ smaller than INT [13]
```
Please note that this is for `bcftools mpileup`.
In some other subtools the `-q` and `-Q` have _swapped_ meaning! (legacy)

**Comment by jodyphelan** [comment]
So the originally they were both set at 0 for Illumina. 
I have now changed mapQ (`-q`) to *8* to avoid multimapping reads with values of 0
baseQ (`-Q`) remains at *0* right now.

**Comment by jodyphelan** [comment]
Just an update. Users can now add/override default parameters using `--calling_params` (pull request: #100). I've run some tests for a small testing dataset consisting of 52 replicates across 10 M/XDR isolates. For all replicates I've tested all combinations for MQs = [0,10,20,30,40,50,60] and BQs = [0,5,10,15,20]. I've observed no difference in the number of SNPs called between different combinations of parameters (see [here](https://gist.github.com/jodyphelan/282ad7b2c69e23108f213670582ef5f3)). I would like to test the effect of these parameters on mixed call frequencies next
